src/hotspot/share/nmt/nmtUsage.cpp

```diff
@@ -52,9 +52,9 @@ void NMTUsage::walk_thread_stacks() {
 }
 
 void NMTUsage::update_malloc_usage() {
-  // Thread critical needed keep values in sync, total area size
+  // Lock needed keep values in sync, total area size
   // is deducted from mtChunk in the end to give correct values.
-  ThreadCritical tc;
+  ChunkPoolLocker lock;
   const MallocMemorySnapshot* ms = MallocMemorySummary::as_snapshot();
```

Comment:
I'd scope this to the as_snapshot call, the subsequent code does not need the locking (lesser chance of accidental recursive locking)

Reply:
done.